pinot-segment-local/src/main/java/org/apache/pinot/segment/local/segment/index/loader/invertedindex/FSTIndexHandler.java

```diff
@@ -74,22 +74,23 @@ public class FSTIndexHandler implements IndexHandler {
   public FSTIndexHandler(SegmentMetadata segmentMetadata, IndexLoadingConfig indexLoadingConfig) {
     _segmentMetadata = segmentMetadata;
     _fstType = indexLoadingConfig.getFSTIndexType();
-    _columnsToAddIdx = new HashSet<>(indexLoadingConfig.getFSTIndexColumns());
+    _columnsToAddIdx = indexLoadingConfig.getFSTIndexColumns();
```

Comment:
(nit) Isn't it probably cleaner and safer (from future changes) to not have `_columnsToAddIdx` as a private member?  Instead just have `_indexLoadingConfig` as a private member and compute `columnsToAddIdx` in the respective methods?

This is the pattern followed in other handlers like BloomFilterHandler,H3IndexHandler.

Reply:
I see BF/H3 hander doesn't hold indexLoadingConfig reference either, but configs for creating BF/H3. 
The handlers in this PR don't have specific configs from indexLoadingConfig either, so not hold its reference.